From 01b43ce07c65d3805cf8c0edc983d095867de8a2 Mon Sep 17 00:00:00 2001
From: Nikhil Benesch <nikhil.benesch@gmail.com>
Date: Mon, 3 Apr 2017 11:19:29 -0700
Subject: [PATCH 6/8] replace sometimes-undefined uint type with unsigned int

Summary:
`uint` is nonstandard and not a built-in type on all compilers; replace it
with the always-valid `unsigned int`. I assume this went unnoticed because
it's inside an `#ifdef ROCKDB_JEMALLOC`.
Closes https://github.com/facebook/rocksdb/pull/2075

Differential Revision: D4820427

Pulled By: ajkr

fbshipit-source-id: 0876561
---
 db/db_impl.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/db/db_impl.cc b/db/db_impl.cc
index 578fe7cb..1aeac56d 100644
--- a/db/db_impl.cc
+++ b/db/db_impl.cc
@@ -601,7 +601,7 @@ static void GetJemallocStatus(void* mstat_arg, const char* status) {
 static void DumpMallocStats(std::string* stats) {
 #ifdef ROCKSDB_JEMALLOC
   MallocStatus mstat;
-  const uint kMallocStatusLen = 1000000;
+  const unsigned int kMallocStatusLen = 1000000;
   std::unique_ptr<char> buf{new char[kMallocStatusLen + 1]};
   mstat.cur = buf.get();
   mstat.end = buf.get() + kMallocStatusLen;
-- 
2.12.1

